### PR TITLE
discord, gateway: update some things to match current API docs

### PR DESCRIPTION
**Pull request by samhza**
Should I make Command.Version its own type like CommandVersion instead of Snowflake?

### CODE REVIEW

File: discord/message.go [hunk 1]
```diff
@@ -147,13 +147,14 @@ const (
 	// name of the thread.
 	ThreadCreatedMessage
 	InlinedReplyMessage
-	ApplicationCommandMessage
+	ChatInputCommandMessage
```
Comment (diamondburned) [inline discord/message.go:150]: Is ChatInputCommandMessage the new ApplicationCommandMessage? Why the rename?

File: discord/permission.go [hunk 2]
```diff
@@ -109,16 +117,15 @@ const (
 		PermissionMoveMembers |
 		PermissionUseVAD |
 		PermissionPrioritySpeaker |
-		PermissionRequestToSpeak
+		PermissionRequestToSpeak |
+		PermissionStartEmbeddedActivities
```
Comment (diamondburned) [inline discord/permission.go:121]: PermissionStartEmbeddedActivities is voice?

File: gateway/events_map.go [hunk 3]
```diff
@@ -73,6 +73,4 @@ var EventCreator = map[string]func() Event{
 
 	"RELATIONSHIP_ADD":    func() Event { return new(RelationshipAddEvent) },
 	"RELATIONSHIP_REMOVE": func() Event { return new(RelationshipRemoveEvent) },
-
-	"APPLICATION_COMMAND_UPDATE": func() Event { return new(ApplicationCommandUpdateEvent) },
```
Comment (diamondburned) [inline gateway/events_map.go:77]: I'm not aware of anything that signifies the removal of this.

**Review by diamondburned — CHANGES_REQUESTED** [review]

File: discord/message.go (hunk 1 above)
Comment (samhza) [reply discord/message.go:150]: Yes. It was renamed when the context menu command message was added to the docs.
https://github.com/discord/discord-api-docs/commit/3973676f437a21714d15338ccab9585c2492847f

File: discord/permission.go (hunk 2 above)
Comment (samhza) [reply discord/permission.go:121]: Yes
> Allows for launching activities (applications with the EMBEDDED flag) in a voice channel
https://discord.com/developers/docs/topics/permissions#permissions-bitwise-permission-flags

File: gateway/events_map.go (hunk 3 above)
Comment (diamondburned) [reply gateway/events_map.go:77]: I'm giving this a go anyway. It doesn't seem to be part of the documented API.

**Review by diamondburned — APPROVED** [review]

File: gateway/events_map.go (hunk 3 above)
Comment (samhza) [reply gateway/events_map.go:77]: https://github.com/discord/discord-api-docs/pull/3691

Comment (samhza) [reply gateway/events_map.go:77]: Yeah it used to be in the documented API but apparently wasn't meant to be there ??? and was removed afterwards.

**Pull request merged**